homeassistant/components/switch/sony_projector.py

```diff
@@ -0,0 +1,102 @@
+"""
+Use SDCP network control to power on/off and check state of Sony projector.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/switch.sony_projector/
+
+Example configuration:
+
+switch:
+  - platform: sony_projector
+    host: "192.168.1.43"
+
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, CONF_NAME, CONF_HOST)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pysdcp==1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Sony Projector'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Connect with serial port and return Acer Projector."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+
+    add_entities([SonyProjector(host, name)], True)
+
+
+class SonyProjector(SwitchDevice):
+    """Represents an Acer Projector as a switch."""
+
+    def __init__(self, host, name):
+        """Init of the Sony projector."""
+        import pysdcp
+        self._sdcp = pysdcp.Projector(host)
+        self._name = name
+        self._host = host
+        self._state = STATE_OFF
+        self._available = False
+        self._attributes = {}
+
+    @property
+    def available(self):
+        """Return if projector is available."""
+        return self._available
+
+    @property
+    def name(self):
+        """Return name of the projector."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return if the projector is turned on."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return state attributes."""
+        return self._attributes
+
+    def update(self):
+        """Get the latest state from the projector."""
+        try:
+            self._state = self._sdcp.get_power()
+            self._available = True
+        except ConnectionRefusedError:
+            _LOGGER.error("Projector connection refused."
+                          " Check IP address (%s) is correct.", self._host)
```

Comment:
The wrong IP address should be handled in `setup_platform()` and here only connection issues.